### ISSUE TITLE
Firebase Hosting Issue - Error 409

### ISSUE THREAD

**Issue by labnol** [firebase/firebase-tools#3327]
My project folder has ~15000 files. When I upload the files to Firebase Hosting using the CLI, it uploads most of the files and then fails when only 30-50 files are left in the queue.

### 1st Attempt
```
i  deploying hosting
i  hosting[xyz]: beginning deploy...
i  hosting[xyz]: found 14914 files in public
⠏  hosting: uploading new files [2711/2754] (98%)
Error: Task 9e811d735dd9a067f8b3cdfed519f58442b75cd1408c9328c03675f5e3f3de8a failed: retries exhausted after 6 attempts
```

### 2nd Attempt
```
i  deploying hosting
i  hosting[xyz]: beginning deploy...
i  hosting[xyz]: found 14914 files in public
⠋  hosting: uploading new files [0/43] (0%)
Error: Task 4e7159c81120864f361eea9caa05dbaf1b6d09e31640218d291f110402c0a1b5 failed: retries exhausted after 6 attempts
```

### 3rd attempt

```
i  deploying hosting
i  hosting[xyz]: beginning deploy...
i  hosting[xyz]: found 14914 files in public
⠙  hosting: uploading new files [0/43] (0%)
Error: Task b6eb2be1cc454118fac3ad5b9c38e93e0a5905d49f846c12ed9c74223bf62b06 failed: retries exhausted after 6 attempts
```

It is unable to download the last 43 files for some reason. I deleted the `.firebase` folder and retried but the same issue is happening.

I added the `--debug` flag and here's the output.

```
[2021-05-01T13:37:51.148Z] <<< [apiv2][body] POST https://upload-firebasehosting.googleapis.com/upload/sites/xyz/versions/fef544579a8665f8/files/131a3baad23bfe2ae3c4c68b715ac13adcc872e7ed5c1f93d0b12f5238b924fd [stream]
[2021-05-01T13:37:51.148Z] [hosting][upload] page-data/28/page-data.json (131a3baad23bfe2ae3c4c68b715ac13adcc872e7ed5c1f93d0b12f5238b924fd) HTTP ERROR 409: [object Headers] Couldn't process request (status=409)
Task 131a3baad23bfe2ae3c4c68b715ac13adcc872e7ed5c1f93d0b12f5238b924fd failed: retries exhausted after 6 attempts
[2021-05-01T13:37:51.149Z] [hosting][upload queue][FINAL] {"max":0,"min":9999999999,"avg":0,"active":42,"complete":1,"success":0,"errored":1,"retried":190,"total":43,"elapsed":22961}
[2021-05-01T13:37:51.151Z] Error: Unexpected error while uploading file.
    at Uploader.uploadHandler (/Users/sites/node_modules/firebase-tools/lib/deploy/hosting/uploader.js:187:19)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at async Queue.executeTask (/Users/sites/node_modules/firebase-tools/lib/throttler/throttler.js:180:22)
    at async Queue.handle (/Users/sites/node_modules/firebase-tools/lib/throttler/throttler.js:96:22)

Error: Task 131a3baad23bfe2ae3c4c68b715ac13adcc872e7ed5c1f93d0b12f5238b924fd failed: retries exhausted after 6 attempts

```

The most common error in the log is `HTTP ERROR 409: [object Headers] Couldn't process request (status=409)`

Please suggest how can I resolve this.

**Comment by labnol** [comment]
On further investigation, I have found that that files that are not getting uploaded have either the `.json` or `.svg` extension.

**Comment by adarshmadrecha** [comment]
I guess your issue is same/similar to https://github.com/firebase/firebase-tools/issues/2126

**Comment by shinycake** [comment]
try this as a temporary fix: https://github.com/firebase/firebase-tools/issues/3335#issuecomment-832927890

**Comment by kmcnellis** [comment]
We're working on a server-side fix for this.  In the meantime, try https://github.com/firebase/firebase-tools/issues/2126#issuecomment-833273969

**Comment by userJulienMoreau** [comment]
The solution that works for me, 
is to create a temporary site on firebase then to deploy the project on this temporary site (I needed 2 build actions then deployment) and it worked.
Then I simply deployed on the right site.
Then I deleted the temporary site.

I tried this solution on 2 different sites and it worked.

It looks like a cache story related to the firebase project... to be continued
